### PR TITLE
fix: pedantic **foo:** and **"word"** emphasis parsing

### PR DESCRIPTION
**Pull request by waiter**
**Marked version:**

18.0.5 / master

**Markdown flavor:** Markdown.pl  / pedantic

## Description

- [Fixes 3938](https://github.com/markedjs/marked/issues/3938)
- Inspired by the fix in [markedjs/marked#3959](https://github.com/markedjs/marked/compare/master...waiter:marked:fix_strong?expand=1), this implementation further optimizes emphasis and strong parsing for broader edge cases, such as:

```
**foo:**bar

**foo**bar

*foo:*bar

_foo:_bar

__foo:__bar

aa**"A"**and**"B"**aa

aa**'A'**and**'B'**aa

**'test'**  
```
image

## Contributor

- [x] Test(s) exist to ensure functionality and minimize regression (if no tests added, list tests covering this PR); or,
- [ ] no tests required for this PR.
- [ ] If submitting new feature, it has been documented in the appropriate places.

## Committer

In most cases, this should be a different person than the contributor.

- [ ] CI is green (no forced merge required).
- [ ] Squash and Merge PR following [conventional commit guidelines](https://www.conventionalcommits.org/).

### CODE REVIEW

**Review by UziTech — APPROVED** [review]
Thanks! this is great 💯

File: src/rules.ts [hunk 1]
```diff
@@ -285,6 +285,13 @@ const _notPunctuationOrSpace = /[^\s\p{P}\p{S}]/u;
 const punctuation = edit(/^((?![*_])punctSpace)/, 'u')
   .replace(/punctSpace/g, _punctuationOrSpace).getRegex();
 
+/** Quote chars for pedantic rule 6 lookbehind; includes both straight and curly forms. */
+const closeQuoteClass = '["\u201c\u201d\u2018\u2019\']';
```
Comment (hongarc) [inline src/rules.ts]: openQuoteClass and closeQuoteClass are asymmetric (” only exists in the latter). If that's intentional, I'd add a test to pin the behavior; otherwise it looks like an easy regression candidate.

File: src/rules.ts [hunk 2]
```diff
@@ -330,7 +345,25 @@ const emStrongRDelimAstGfm = edit(emStrongRDelimAstCore, 'gu')
...
+// Pedantic: rule 3 requires whitespace before left delim; rule 6 allows punct before close (e.g. `:**bar`).
+const emStrongRDelimAstPedanticCore =
+  '^[^_*]*?__[^_*]*?\\*[^_*]*?(?=__)' // Skip orphan inside strong
++ '|[^*]+(?=[^*])' // Consume to delim
++ '|(?!\\*)punct(\\*+)(?=[\\s]|$)' // (1) #*** can only be a Right Delimiter
++ '|notPunctSpace(\\*+)(?!\\*)(?=punctSpace|$)' // (2) a***#, a*** can only be a Right Delimiter
++ '|(?!\\*)[\\s](\\*+)(?=notPunctSpace)' // (3) ***a can only be Left Delimiter (whitespace required; not #***a)
++ '|[\\s](\\*+)(?!\\*)(?=punct)' // (4) ***# can only be Left Delimiter
++ '|(?!\\*)punct(\\*+)(?!\\*)(?=punct)' // (5) #***# can be either Left or Right Delimiter
++ '|closeQuoteLookbehind(?:(?!\\*)punct|notPunctSpace)(\\*+)(?!\\*)(?=notPunctSpace)'; // (6) a***a, #***a, and :** (e.g. **foo:**bar)
```
Comment (hongarc) [inline src/rules.ts]: This changes pedantic parsing for some *"a*a-style edge cases. The new behavior may well be preferable, but since it's a silent regression, I'd add a regression test to explicitly pin the intended behavior

**Comment by waiter** [comment]
Updated! Expanded coverage for more strong/emphasis patterns, tuned Rule 6, and added test cases.

@UziTech  @hongarc  PTAL when you get a chance!

**Review by UziTech — APPROVED** [review]
